comp/otelcol/otlp/collector.go

```diff
@@ -58,10 +104,23 @@ func getComponents(s serializer.MetricSerializer, logsAgentChannel chan *message
 	if err != nil {
 		errs = append(errs, err)
 	}
+	addr := fmt.Sprintf("http://localhost:%s/v0.6/stats", pkgconfigsetup.Datadog.GetString("apm_config.receiver_port"))
 
+	hname, err := hostname.Get(context.TODO())
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	var extraTags []string
+
+	// if the server is running in a context where static tags are required, add those
+	// to extraTags.
+	if tags := util.GetStaticTagsSlice(context.TODO()); tags != nil {
+		extraTags = append(extraTags, tags...)
+	}
 	exporterFactories := []exporter.Factory{
 		otlpexporter.NewFactory(),
-		serializerexporter.NewFactory(s),
+		serializerexporter.NewFactory(s, &tagEnricher{cardinality: collectors.LowCardinality}, addr, extraTags, hname),
```

Comment:
We should call the hostname function each time we need it, in case in the future it can change at runtime